packages/core/src/ToastProvider/ToastProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { createContext, useCallback, useContext, useEffect, useState } from 'react'
+import { createPortal } from 'react-dom'
+import { Absolute } from '../Absolute'
+import { Flex } from '../Flex'
+import { Toast, IToastProps } from '../Toast'
+import { ThemeProvider } from '../ThemeProvider'
+import { Box } from '../Box'
+
+interface IToastOptions extends IToastProps {
+  removed?: boolean
+}
+
+interface IToastContextProps {
+  addToast: (options: IToastOptions) => void
+  removeToast: (id: number) => void
+}
+
+/* istanbul ignore next */
+export const ToastContext = createContext<IToastContextProps>({ addToast: () => {}, removeToast: () => {} })
+
+export const useToast = () => {
+  return useContext(ToastContext)
+}
+
+interface IToastProvider {
+  children: React.ReactNode
+  domRootId?: string
+  lifespan?: number
+  maxToasts?: number
+}
+
+let id = 0
+
+function ToastProvider({ children, domRootId = 'root', lifespan, maxToasts = 3 }: IToastProvider) {
+  const [toasts, setToasts] = useState<IToastOptions[]>([])
+
+  const addToast = useCallback((options: IToastOptions) => {
+    setToasts((prevToasts) => [...prevToasts, { ...options, id: id++ }])
+  }, [])
+
+  const removeToast = useCallback((id: number) => {
+    setToasts((prevToasts) =>
+      prevToasts.map((toast) => (toast.id === id ? { ...toast, removed: true } : toast))
+    )
+  }, [])
+
+  useEffect(() => {
+    toasts.forEach((toast) => {
+      if (toast.removed) {
+        setTimeout(() => {
+          setToasts((prevToasts) => prevToasts.filter((t) => t.id !== toast.id))
+        }, 200)
+      }
+    })
+  }, [toasts, setToasts])
+
+  const toastsToRender = toasts.slice(0, maxToasts)
+
+  return (
+    <ToastContext.Provider value={{ addToast, removeToast }}>
+      {children}
+      {createPortal(
+        <ThemeProvider>
+          <Absolute bottom={20} width='100%'>
+            <Flex justifyContent='center' width='100%'>
+              <Flex flexDirection='column-reverse' justifyContent='center' minWidth='300px'>
+                {toastsToRender.map((toast, idx) => {
+                  return (
+                    <Box mt={3}>
+                      <Toast key={idx} lifespan={lifespan} onRemoveClick={removeToast} {...toast} />
+                    </Box>
```

Comment:
I added a `<Box mt={3}>` because `<Toast key={idx} lifespan={lifespan} onRemoveClick={removeToast} {...toast} mt={3}/>` gives me this:

<img width="412" alt="Screenshot 2023-04-10 at 9 45 51 AM" src="https://user-images.githubusercontent.com/93283233/230925514-d8f31e93-112a-4be3-92be-b443930a8fb4.png">